verifyFiles.py

```diff
@@ -144,19 +147,27 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics):
                 for subject in conditions_dict[c].keys():
                 # Here we assume that both conditions will have the same set of subjects
                     files_are_different=False
-                    if(conditions_dict[c][subject][file_name].st_size != conditions_dict[d][subject][file_name].st_size):
+		    abs_path_c=os.path.join(root_dir,c,subject,file_name)
+                    abs_path_d=os.path.join(root_dir,d,subject,file_name)
+		    if checksums_from_file_dict:
+                            file_name_checksum = subject+"/"+file_name
+		    	    if (checksums_from_file_dict[c][subject][file_name_checksum] != checksums_from_file_dict[d][subject][file_name_checksum]):
+                                diff[key][file_name]+=1
+                                files_are_different=True
+		    elif conditions_dict[c][subject][file_name].st_size != conditions_dict[d][subject][file_name].st_size :
                         #diff[key][file_name]["binary_content"]+=1
 			diff[key][file_name]+=1
                         files_are_different=True
-                    else:
-                        # File sizes are identical: compute the checksums
-                        abs_path_c=os.path.join(root_dir,c,subject,file_name)
-                        abs_path_d=os.path.join(root_dir,d,subject,file_name)
-                        if checksum(abs_path_c) != checksum(abs_path_d): # TODO:when they are multiple conditions, we will compute checksums multiple times.
-                                                                         # We should avoid that.
-			    diff[key][file_name]+=1
-                            files_are_different=True
+	   	    elif checksum(abs_path_c) != checksum(abs_path_d):# TODO:when there are multiple conditions, we will compute checksums multiple times.We should avoid that.
+		        #Condition below makes sure that the checksums in the file after processing and in local are equal
+			diff[key][file_name]+=1
+                        files_are_different=True
+
                     if files_are_different:
+			if checksums_from_file_dict and (checksum(abs_path_c) != checksums_from_file_dict[c][subject][file_name_checksum]) and checksum_after_file_path not in file_name_checksum:
+                            log_error("Checksum of\"" + abs_path_c + "\"in checksum file is different from what is computed here.")
+                        if checksums_from_file_dict and (checksum(abs_path_d) != checksums_from_file_dict[d][subject][file_name_checksum]) and checksum_after_file_path not in file_name_checksum:
+                            log_error("Checksum of\"" + abs_path_d + "\"in checksum file is different from what is computed here.")
```

Comment:
That works but the code could be factorized a little bit and be better commented (it's not straightforward).